app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -1,55 +1,531 @@
+                val imageResource = signResources.getSpeedSignResource(
+                        UiSign.WithNumber(
+                                signType = UiSign.SignType.SpeedLimit,
+                                signNumber = UiSign.SignNumber.fromNumber(roadRestrictions.speedLimits.car.max)
+                        ),
+                        speed = lastSpeed,
+                        country = country
+                )
+
+                speed_limit_current.animate().cancel()
+                speed_limit_next.animate().cancel()
+
+                speed_limit_current.apply {
+                    show()
+                    translationY = 0f
+                    alpha = 1f
+                    animate()
+                            .translationY(speedLimitTranslation / 2)
+                            .alpha(0f)
+                            .scaleX(0.5f)
+                            .scaleY(0.5f)
+                            .setDuration(500L)
+                            .setListener(
+                                    object : Animator.AnimatorListener {
+                                        override fun onAnimationRepeat(animation: Animator?) {}
+
+                                        override fun onAnimationEnd(animation: Animator?) {
+                                            setImageResource(imageResource)
+                                            translationY = 0f
+                                            alpha = 1f
+                                            scaleX = 1f
+                                            scaleY = 1f
+                                            speed_limit_next.hide()
+                                        }
+
+                                        override fun onAnimationCancel(animation: Animator?) {}
+
+                                        override fun onAnimationStart(animation: Animator?) {}
+                                    }
+                            )
+                            .setInterpolator(AccelerateDecelerateInterpolator())
+                            .start()
+                }
+
+                if (roadRestrictions.speedLimits.car.max != 0f) {
+                    speed_limit_next.apply {
+                        translationY = -speedLimitTranslation
+                        setImageResource(imageResource)
+                        show()
+                        animate().translationY(0f)
+                                .setDuration(500L)
+                                .setInterpolator(AccelerateDecelerateInterpolator())
+                                .start()
+                    }
+                } else {
+                    speed_limit_next.hide()
+                }
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+        super.onCreate(savedInstanceState)
+
+        if (!SystemInfoUtils.isVisionSupported()) {
```

Comment:
```suggestion
        if (SystemInfoUtils.isVisionSupported().not()) {
```

Reply:
@alexander-kulikovskii IMHO it isn't really looking better. WDYT guys?

Reply:
it’s easier not to notice `!` sign than `.not()` when refactor this code
IMHO

Reply:
Thing is - `!condition` is the default (and expected by everyone, not only kotlin-ists) negation expression, while `not()` is not

Reply:
IMO not() is more verbose and ! is easy to miss 
People should be familiar with Kotlin if they plan to read code anyway

Reply:
used not() for now, but we can move this discussion to slack

Reply:
Independent of the final solution I don't think its a good idea to change negation in this _single_ case, it'll confuse us and other's even further

Reply:
okay, will wait for all team decisions or move discussion to slack to speedup process

Reply:
not  is my 🥇